src/main/java/com/google/cloud/tools/managedcloudsdk/install/Extractor.java

```diff
@@ -81,4 +82,10 @@ ExtractorProvider getExtractorProvider() {
   private void cleanUp(final Path target) throws IOException {
     MoreFiles.deleteRecursively(target, RecursiveDeleteOption.ALLOW_INSECURE);
   }
+
+  static boolean isTargetInsideDestination(Path destination, Path target) throws IOException {
+    String canonicalDestination = destination.toFile().getCanonicalPath();
+    String canonicalTarget = target.toFile().getCanonicalPath();
+    return canonicalTarget.startsWith(canonicalDestination + File.separator);
```

Comment:
String prefix checking won't catch "/usr/libexec" vs "/usr/lib".  Convert to a `Path` and use `Path.startsWith(Path)`?

Reply:
We are good because we add `File.separator`. I just consulted the example code for fix coming from the very folks who discovered this vulnerability, so I think we're safe. You might still argue using `Path` seems ideal, but when I tried it, I realized that `Path.startsWith()` will work differently for the case where two paths are identical. So, I guess following the example code is reasonable.

Reply:
Ok, I missed the `File.separator`.